lib/priorityQueue.js

```diff
@@ -62,25 +48,26 @@ export default function(worker, concurrency) {
                 q.drain();
             });
         }
+
+        var nextNode = q._tasks.head;
+        while (nextNode && priority >= nextNode.priority) {
+            nextNode = nextNode.next;
+        }
+
         arrayEach(data, function(task) {
             var item = {
                 data: task,
                 priority: priority,
                 callback: typeof callback === 'function' ? callback : noop
             };
 
-            q.tasks.splice(_binarySearch(q.tasks, item, _compareTasks) + 1, 0, item);
-
-            setImmediate(q.process);
```

Comment:
note this change, moved process outside of the loop. I think this is the correct behaviour